src/js/media-chrome-button.ts

```diff
@@ -92,23 +94,29 @@ template.innerHTML = /*html*/ `
  * @cssproperty --media-button-icon-transform - `transform` of button icon.
  * @cssproperty --media-button-icon-transition - `transition` of button icon.
  */
-class MediaChromeButton extends globalThis.HTMLElement {
-  #mediaController;
+class MediaChromeButton extends CustomElement {
```

Comment:
The `CustomElement` class does *not* use the server safe `globalThis`. This is because when that version of the class is used, typescript no longer sees the components as extensions of `HTMLElement`, and marks any reference to `this` in the class as invalid. Some investigation is needed into how to get around these errors, or if there is a better way to do "sever safe globals".

Reply:
I had some fights against this while I was working on the media-store abstraction. I'm starting to suspect we should do this iteratively, starting with things like constants and server-safe-globals and working our way out. We're definitely in a tricky TS situations with the node/server vs. browser straddling.

Reply:
What is the exact problem that `server-safe-globals.js` solves? (is there an issue and/or PR?)

Reply:
Ok a sub-optimal compromise I'm experimenting with in a local branch is "lying" with our types in our our server-safe-globals definitions. Ideally we'll follow up to make sure that we don't end up with runtime errors for valid server-side usage, but this feels like the right compromise to accelerate/unblock the TS migration effort.

I'll push up a branch for you to see the general approach there. Might make sense for us to touch base on how to combine your effort with my changes.

Reply:
I just ran a test, removed server-safe-globals completely from the source files, only use it in `build/react.js`, and both example apps build without issue.

Reply:
https://github.com/littlespex/media-chrome/pull/4

Reply:
I promise these are necessary shims.